_posts/2019-04-30-accessing-kafka-part-3.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title:  "Accessing Kafka: Part 3 - OpenShift Routes"
+date: 2019-04-30
+author: jakub_scholz
+---
+
+In the third part of this blog post series we will look at exposing Kafka using OpenShift Routes.
+This post will explain how routes work and how they can be used with Kafka.
+Routes are available only on OpenShift.
+But if you are a Kubernetes user, don't be sad, one of the next part will be about using Kubernetes Ingress which is similar to OpenShift routes.
+
+<!--more-->
+
+_This post is part of a bigger series about different ways to access a Kafka cluster powered by Strimzi.
+The other parts published so far are:_
+
+* _[Part 1 - Introduction](https://strimzi.io/2019/04/17/accessing-kafka-part-1.html)_
+* _[Part 2 - Node Ports](https://strimzi.io/2019/04/23/accessing-kafka-part-2.html)_
+* _Part 3 - OpenShift Routes (this post)_
+
+# OpenShift Routes
+
+Routes are an OpenShift concept for exposing services to the outside of the OpenShift platform.
+Routes handle both data routing as well as DNS resolution.
+DNS resolution is usually handled using [wildcard DNS entries](https://en.wikipedia.org/wiki/Wildcard_DNS_record).
+That allows OpenShift to assign each route its own DNS name which is based on the wildcard entry.
+Users do not have to anything special to handle the DNS records.
+But don't worry, when you don't own any domains where you can setup the wildcard entires, it can use services such as [nip.ip](https://nip.io/) for the wildcard DNS routing.
+Data routing is done using the [HAProxy](https://www.haproxy.org) load balancer which serves as the router behind the routes.
```

Comment:
"the router behind the routes" sounds a bit confusing. I don't know whether "the router behind the domain names" is really any better though.